group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -739,4 +741,15 @@ private static Integer decValue(String key, Integer value) {
     private static Integer incValue(String key, Integer value) {
         return value == null ? 1 : value + 1;
     }
+
+    /**
+     * @return the group formatted as a list group response.
+     */
+    public ListGroupsResponseData.ListedGroup asListedGroup() {
```

Comment:
Don't we need to also implement this for `GenericGroup`?

Reply:
> Don't we need to also implement this for `GenericGroup`?

GenericGroup has implemented the asListedGroup method. In fact, ConsumerGroup has this method first. In order to implement ListGroups, I abstracted it into an interface and implemented it in GenericGroup.